### PR TITLE
Release documentation

### PR DESCRIPTION
**Pull request by boek**

### CODE REVIEW

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 1]
```diff
@@ -0,0 +1,49 @@
...
+
+## String freeze
+- [ ] Make sure all issues with label "strings needed" have strings assigned and the label updated to "strings approved"
+- [ ] Pre-land all strings for features that have not been implemented yet
+
+## Soft code freeze [Wednesday, 2nd week of sprint]
+- [ ] Create a branch for the *current* milestone and protect it through Settings on the repo (need admin privileges). After that master is tracking the next milestone.
+- [ ] [Create an issue](https://github.com/mozilla-mobile/fenix/issues/new?template=release_checklist.md&title=Releng+for+) in the *upcoming* milestone: "Releng for [release]".
+- [ ] Go through the list of bugs closed during this sprint and make sure all they're all added to the correct milestone.
+- [ ] Add either `eng:qa:neede` flags on each issue that still needs it.
```
Comment (colintheshots) [inline .github/ISSUE_TEMPLATE/release_checklist.md]: typo here

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 2]
```diff
@@ -0,0 +1,49 @@
...
+## During Product Integrity
+
+- [ ] Check Google Play for new crashes. File issues and triage.
+- [ ] If bugs are considered release blocker then fix them on master and the milestone branch (cherry-pick / uplift)
+- [ ] If needed tag a new RC version (e.g. v1.0-RC2) and follow the submission checklist again.
+
+## Release
+
+- [ ] Create a release request in Bugzilla. You can clone [this issue](https://bugzilla.mozilla.org/show_bug.cgi?id=1571967) and NI someone from release management.
+- [ ] Release management promotes the build from beta to the release channel. For minor releases this can happen at any time during the day. Major releases often need to be synchronized with other marketing activities (e.g. blog postings). Releases are rolled out to 99% of the population (otherwise the rollout can't be stopped).
```
Comment (colintheshots) [inline .github/ISSUE_TEMPLATE/release_checklist.md]: Nit: This doesn't seem like a checklist item that can get checked off currently. It should be "Wait until release management successfully promotes the build from beta to release and removes NI from the Bug..." It's also possible, though unlikely, that this step can fail for build-related reasons that will need to be resolved, like improper versionCodes.

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 3]
```diff
@@ -0,0 +1,49 @@
...
+
+## After the release
+
+- [ ] After some days release management sets the roll out to 100%
```
Comment (colintheshots) [inline .github/ISSUE_TEMPLATE/release_checklist.md]: This step may be checking to ensure the roll out gets increased to 100%. As stated, the checklist item assumes that release management is responsible for doing so and so the checklist item has no bearing on the reader.

File: .github/ISSUE_TEMPLATE/release_checklist.md [hunk 4]
```diff
@@ -0,0 +1,49 @@
...
+- [ ] Check whether there are new crashes that need to be filed.
+- [ ] Tag the last and released RC version additionally with the tag of the release (v1.0-RC2 -> v1.0)
+- [ ] Upload signed APKs to the [release page](https://github.com/mozilla-mobile/fenix/releases)
```
Comment (colintheshots) [inline .github/ISSUE_TEMPLATE/release_checklist.md]: This is debatable, but these last two items could probably be first in this section. We don't want to wait until a week after the release to hopefully remember to tag the release and add APKs to Github. If something causes us to roll back, we'll want to create a new minor version for the fix anyway. We've been forgetting to add APKs entirely and in-house testers are finding it very difficult.

File: .github/ISSUE_TEMPLATE/release_checklist.md (hunk 2 above)
Comment (boek) [reply .github/ISSUE_TEMPLATE/release_checklist.md]: Hmm good point. This step was mostly to make sure that we go from xx% to 100% but that definitely didn't get communicated well

**Pull request merged**